auction-server/src/opportunity/api.rs

```diff
@@ -202,6 +202,7 @@ pub async fn get_opportunities(
     }
 }
 
+const MEMO_MAX_LENGTH: usize = 100;
```

Comment:
decide a reasonable value here

Reply:
is this to prevent DoSooor quote requestoors from making a bunch of illegitimate quote requests with too-long-to-be-included-in-a-transaction memos?

Reply:
if they pass a very long memo, they will just get 404 because the searcher transactions will be too big. 
it's better to fail explicitly here so the user understands the reason for not getting any quotes is the length of the memo.